src/envoy/http/authn_wasm/config.h

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "extensions/common/wasm/null/null.h"
+
+#include "envoy/config/filter/http/authn/v2alpha1/config.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+namespace AuthnWasm {
+
+using istio::envoy::config::filter::http::authn::v2alpha1::FilterConfig;
+using StringView = absl::string_view;
+using Common::Wasm::Null::NullVmPluginRootRegistry;
+using Common::Wasm::Null::Plugin::Context;
+using Common::Wasm::Null::Plugin::ContextFactory;
+using Common::Wasm::Null::Plugin::RootContext;
+using Common::Wasm::Null::Plugin::RootFactory;
+using Common::Wasm::Null::Plugin::WasmData;
+
+// PluginRootContext is the root context for all streams processed by the
```

Comment:
is this comment accurate? i saw the `wasm_filter.md` is documented this is a per vm instance.

Reply:
I copied from other files, why is this not accurate? It talks about streams and thread but doesn't mention VMs?

Reply:
Yes.`wasm_filter.md` references the VM not the thread. @mandarjog and @jplevyak in last weekly meeting, we talked about the mapping of VM/worker threads, I can't remember the details though to confirm if it's 1:1 mapping. This seems closed to existing Envoy filter context, not sure how much applied to Wasm one.

> At bootstrap time, a root context is created. The root context
has the same lifetime as the VM/runtime instance and acts as a target
for any interactions which happen at initial setup. It is also used for
interactions that outlive a request.